Documentation/Willow 4.0 Migration Guide.md

```diff
@@ -0,0 +1,63 @@
+# Willow 4.0 Migration Guide
+
+Willow 4.0 is the latest major release of Willow, a powerful, yet lightweight logging library for iOS, macOS, tvOS and watchOS written in Swift. 
+As a major release, following Semantic Versioning conventions, 4.0 introduces several API-breaking changes that one should be aware of.
+
+This guide is provided in order to ease the transition of existing applications using Willow 3.x to the latest APIs.
+
+## Requirements
+
+Willow 4.0 officially supports iOS 9.0+, macOS 10.11+, tvOS 9.0+, watchOS 2.0+, Xcode 9.0+ and Swift 4.0+. 
+If you'd like to use Willow in a project targeting Xcode 8.3 and Swift 3.1, use the latest tagged 3.x release.
+
+---
+
+## Breaking API Changes
+
+Willow 4.0 contains only two breaking API changes that are very minor, but needed to be corrected. 
+Most of your Willow code will be able to remain the same.
+
+### LogModifier
+
+The `LogMessageModifier` type has been renamed to `LogModifier`
+
+### LogWriter
+
+The `LogMessageWriter` type has been renamed to `LogWriter`
+
+### LoggerConfiguration
+
+The `LoggerConfiguration` type has been removed as a part of making Willow easier to setup and use. 
+`LogModifier`s are now passed into `LogWriter`s when they are initialized and are not associated with specific log levels.
+
+Creating a new `Logger` instance is now as simple as initializing with a set of `LogLevel`s, `LogWriters` and the `ExecutionMethod`.
+
+```swift
+let log =  Logger(logLevels: .all, writers: [ConsoleWriter(modifiers: [TimestampModifier()])])
+```
+
+## New Functionality
+
+### Messages
+In addition to logging strings, Willow now support logging attributed data in the form of `LogMessages`. 
+`LogMessages` are structured data with a name and a dictionary of attributes. 
+Willow declares the `LogMessages` protocol which frameworks and applications can use as the basis for concrete implementations.
```

Comment:
This reads as though the protocol is named `LogMessages`, not `LogMessage`. Should rephrase a bit so that isn't the case.